test_conformance/compiler/test_feature_macro.cpp

```diff
@@ -0,0 +1,398 @@
+
+    test_error(
+        error,
+        "clGetDeviceInfo asking for CL_DEVICE_OPENCL_C_FEATURES failed.\n");
+    if (config_size == 0)
+    {
+        log_info("Empty list of CL_DEVICE_OPENCL_C_FEATURES returned by "
+                 "clGetDeviceInfo on this device.\n");
+    }
+    else
+    {
+        int vec_elements = config_size / sizeof(cl_name_version);
+        vec_device_feature.resize(vec_elements);
+        error = clGetDeviceInfo(deviceID, CL_DEVICE_OPENCL_C_FEATURES,
+                                config_size, vec_device_feature.data(), 0);
+        test_error(
+            error,
+            "clGetDeviceInfo asking for CL_DEVICE_OPENCL_C_FEATURES failed.\n");
+    }
+    for (auto each_f : vec_device_feature)
+    {
+        vec_device_feature_names.push_back(each_f.name);
+    }
+    sort(vec_to_cmp.begin(), vec_to_cmp.end());
+    sort(vec_device_feature_names.begin(), vec_device_feature_names.end());
+
+    bool result =
+        std::equal(vec_device_feature_names.begin(),
+                   vec_device_feature_names.end(), vec_to_cmp.begin());
+    if (result)
+    {
+        log_info("Comparison list of features - passed\n");
+    }
+    else
+    {
+        log_info("Comparison list of features - failed\n");
+        error = TEST_FAIL;
+    }
+    log_info(
+        "Supported features based on CL_DEVICE_OPENCL_C_FEATURES API query:\n");
+    for (auto each_f : vec_device_feature_names)
+    {
+        log_info("%s\n", each_f.c_str());
+    }
+
+    log_info("\nSupported features based on queries to API/compiler :\n");
+    for (auto each_f : vec_to_cmp)
+    {
+        log_info("%s\n", each_f.c_str());
+    }
+
+    return error;
+}
+
+#define NEW_FEATURE_MACRO_TEST(feat)                                           \
+    test_macro_name = "__opencl_c_" #feat;                                     \
+    error |= test_feature_macro_##feat(deviceID, context, test_macro_name,     \
+                                       supported);                             \
+    if (supported) supported_features_vec.push_back(test_macro_name);
+
+
+int test_features_macro(cl_device_id deviceID, cl_context context,
+                        cl_command_queue queue, int num_elements)
+{
+    cl_int error = CL_SUCCESS;
+    bool supported = false;
+    std::string test_macro_name = "";
+    std::vector<std::string> supported_features_vec;
+    NEW_FEATURE_MACRO_TEST(program_scope_global_variables);
+    NEW_FEATURE_MACRO_TEST(3d_image_writes);
+    NEW_FEATURE_MACRO_TEST(atomic_order_acq_rel);
+    NEW_FEATURE_MACRO_TEST(atomic_order_seq_cst);
+    NEW_FEATURE_MACRO_TEST(atomic_scope_device);
+    NEW_FEATURE_MACRO_TEST(atomic_scope_all_devices);
+    NEW_FEATURE_MACRO_TEST(device_enqueue);
+    NEW_FEATURE_MACRO_TEST(generic_address_space);
+    NEW_FEATURE_MACRO_TEST(pipes);
+    NEW_FEATURE_MACRO_TEST(read_write_images);
+    NEW_FEATURE_MACRO_TEST(subgroups);
+    NEW_FEATURE_MACRO_TEST(work_group_collective_functions);
```

Comment:
Need to add the following features:
- `__opencl_c_images`
- `__opencl_c_fp64`
- `__opencl_c_int64`